substrate/frame/state-trie-migration/src/lib.rs

```diff
@@ -618,8 +679,10 @@ pub mod pallet {
 
 			// ensure that the migration witness data was correct.
 			if real_size_upper < task.dyn_size {
+				T::Currency::hold(&HoldReason::SlashForContinueMigrate.into(), &who, deposit)?;
 				// let the imbalance burn.
```

Comment:
@kianenigma 
Why do we just burn slashed imbalance here?
Shouldnt we better add here `OnUnbalanced::on_unbalanced` handler?